### PR TITLE
Fix: add extra escaping to form grid shortcode attributes

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This adds extra escaping to the form grid shortcode attribute rendering to prevent XSS.

## Affects

The form grid shortcode

## Visuals

N/A

## Testing Instructions

Make sure the form grid shortcode works as expected.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204137775197143

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@kjohnson @pauloiankoski this is ready for review.

File: templates/shortcode-form-grid.php [hunk 1]
```diff
@@ -198,9 +206,9 @@ static function($term)use($tag_text_color,$tag_bg_color){
                         : '#000000';
                     ?>
                     <button style="text-decoration-color: <?php
-                    echo $button_text_color; ?>">
+                    echo esc_attr($button_text_color); ?>">
                                     <span style="color: <?php
-                                    echo $button_text_color; ?>">
+                                    echo esc_attr($button_text_color); ?>">
                                         <?php
                                         echo $button_text ?: __('Donate', 'give'); ?>
```
Comment (pauloiankoski) [inline templates/shortcode-form-grid.php:213]: `$button_text` should be escaped.

**Review by pauloiankoski — CHANGES_REQUESTED** [review]

File: templates/shortcode-form-grid.php (hunk 1 above)
Comment (jonwaldstein) [reply templates/shortcode-form-grid.php]: resolved e8806385

**Review by pauloiankoski — APPROVED** [review]
Looks good. Ready for QA.

**Comment by kjohnson** [comment]
I'm re-running failed jobs (looks like a date comparison is off by a couple of seconds).

**Review by Benunc — APPROVED** [review]
Manually confirmed that the malicious shortcode (a) currently breaks on latest stable and (b) is fixed by this PR.

Also ran all automated tests. 

:shipit:

**Pull request merged**